src/util/metricsbackend.cpp

```diff
@@ -0,0 +1,111 @@
+/*
+Minetest
+Copyright (C) 2013-2020 Minetest core developers team
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation; either version 2.1 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along
+with this program; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+#include "metricsbackend.h"
+#if USE_PROMETHEUS
+#include <prometheus/counter.h>
+#include <prometheus/gauge.h>
+#include "log.h"
+#include "settings.h"
+#endif
+
+MetricCounterPtr MetricsBackend::addCounter(
+		const std::string &name, const std::string &help_str)
+{
+	return std::make_shared<SimpleMetricCounter>(name, help_str);
+}
+
+MetricGaugePtr MetricsBackend::addGauge(
+		const std::string &name, const std::string &help_str)
+{
+	return std::make_shared<SimpleMetricGauge>(name, help_str);
+}
+
+#if USE_PROMETHEUS
+
+PrometheusMetricCounter::PrometheusMetricCounter(const std::string &name,
+		const std::string &help_str,
+		std::shared_ptr<prometheus::Registry> registry) :
+		MetricCounter(),
+		m_family(prometheus::BuildCounter().Name(name).Help(help_str).Register(
+				*registry)),
+		m_counter(m_family.Add({}))
+{
+}
+
+void PrometheusMetricCounter::increment(double number)
+{
+	m_counter.Increment(number);
+}
+
+double PrometheusMetricCounter::get() const
+{
+	return m_counter.Value();
+}
+
+PrometheusMetricGauge::PrometheusMetricGauge(const std::string &name,
+		const std::string &help_str,
+		std::shared_ptr<prometheus::Registry> registry) :
+		MetricGauge(),
+		m_family(prometheus::BuildGauge().Name(name).Help(help_str).Register(
+				*registry)),
+		m_gauge(m_family.Add({}))
+{
+}
+
+void PrometheusMetricGauge::increment(double number)
+{
+	m_gauge.Increment(number);
+}
+void PrometheusMetricGauge::decrement(double number)
+{
+	m_gauge.Decrement(number);
+}
+void PrometheusMetricGauge::set(double number)
+{
+	m_gauge.Set(number);
+}
+double PrometheusMetricGauge::get() const
+{
+	return m_gauge.Value();
+}
+
+PrometheusMetricsBackend::PrometheusMetricsBackend() : MetricsBackend()
+{
+	std::string addr;
+	if (!g_settings->getNoEx("prometheus_listener_address", addr))
+		addr = "127.0.0.1:8080";
```

Comment:
There is a default value in `defaultsettings.cpp` so this shouldn't be needed.

Reply:
okay, dropping